autocti/extract/two_d/serial/abstract.py

```diff
@@ -0,0 +1,129 @@
+
+import autoarray as aa
+
+from autocti.extract.two_d.abstract import Extract2D
+
+
+class Extract2DSerial(Extract2D):
+    @property
+    def binning_axis(self) -> int:
+        """
+        The axis over which binning is performed to turn 2D serial data (E.g. an EPER) into 1D data.
+
+        For a serial extract `axis=0` such that binning is performed over the rows containing the EPER.
+        """
+        return 0
+
+    def median_list_from(
+        self, array: aa.Array2D, pixels: Tuple[int, int]
+    ) -> List[float]:
+        """
+        Returns the median values of the `Extract2D` object's corresponding region, where the median is taken over
+        all rows of the region(s).
+
+        To describe this function we will use the example of estimating the median of the charge lines in charge
+        injection data, by taking the median over the rows of charge injection (e.g. each median is anti-aligned with
+        the FPR).
+
+        By taking the median of values of the charge injection regions (after accounting for those which have had
+        electrons captured and relocated due to CTI), we can therefore estimate whether there is row-to-row variation
+        in the charge injection.
+
+        This function does this for every rows of every charge injection. If multiple charge injections are performed,
+        the median of all charge regions is taken.
+
+        For example, if there are 3 charge injection regions, this function returns a list where each value
+        estimates the charge injection normalization of a given charge injection region. The size of this list
+        is therefore the number of charge injection rows.
+
+        The function `median_list_of_lists_from` performs the median over each individual charge injection
+        region in each row. It therefore estimates multiple injection normalizations per row. for each individual
+        charge region. Which function one uses depends on the properties of the charge injection on the instrumentation.
+
+        Parameters
+        ----------
+        array
+            The charge injection image from which the charge injection normalizations are estimated.
+        pixels
+            The row pixel index to extract the FPR between (e.g. `pixels=(0, 3)` extracts the 1st, 2nd and 3rd
+            FPR rows). To remove the 10 leading pixels which have lost electrons due to CTI, an input such
+            as `pixels=(10, 30)` would be used.
+        """
+        median_list = []
+
+        arr_list = [
+            np.ma.array(data=array.native[region.slice], mask=array.mask[region.slice])
+            for region in self.region_list_from(pixels=pixels)
+        ]
+
+        for row_index in range(arr_list[0].shape[0]):
+
+            for i, array_2d in enumerate(arr_list):
+
+                if i == 0:
+                    arr = array_2d[row_index, :]
+                else:
+                    arr = np.concatenate((arr[:], array_2d[row_index, :]))

```

Comment:
Same here. Also by using stack you can avoid this weird index based disjunct within the loop